internal/homing/repository/timescale/init.go

```diff
@@ -0,0 +1,68 @@
+package timescale
+
+import (
+	"fmt"
+
+	_ "github.com/jackc/pgx/v5/stdlib" // required for SQL access
+	"github.com/jmoiron/sqlx"
+	migrate "github.com/rubenv/sql-migrate"
+)
+
+// Config defines the options that are used when connecting to a TimescaleSQL instance
+type Config struct {
+	Host        string `env:"MF_TIMESCALE_HOST"   envDefault:"localhost"`
+	Port        string `env:"MF_TIMESCALE_PORT"   envDefault:"5432"`
+	User        string `env:"MF_TIMESCALE_USER"`
+	Pass        string `env:"MF_TIMESCALE_PASSWORD"`
+	Name        string `env:"MF_TIMESCALE_DB_NAME"`
+	SSLMode     string `env:"MF_TIMESCALE_SSL_MODE"  envDefault:"disable"`
+	SSLCert     string `env:"MF_TIMESCALE_SSL_CERT"`
+	SSLKey      string `env:"MF_TIMESCALE_SSL_KEY"`
+	SSLRootCert string `env:"MF_TIMESCALE_SSL_ROOT_CERT"`
+}
+
+// Connect creates a connection to the TimescaleSQL instance and applies any
+// unapplied database migrations. A non-nil error is returned to indicate
+// failure.
+func Connect(cfg Config) (*sqlx.DB, error) {
```

Comment:
Use this approach https://github.com/mainflux/mainflux/blob/master/users/postgres/init.go